src/openms/source/FORMAT/SwathFile.cpp

```diff
@@ -222,10 +223,10 @@ namespace OpenMS
 
     startProgress(0, 1, "Loading metadata file " + file);
     boost::shared_ptr<PeakMap > experiment_metadata(new PeakMap);
-    MzXMLFile f;
+    FileHandler f;
```

Comment:
hmm probably works for mzML and mzXML, for the others I think the options don't work at all.
One would also need to carefully check if something mzXML specific (e.g., MetaValues) are used.
In general, I am a bit divided regarding having a generic FileHandler. It first sounds like a great idea to abstract over different input files. But the devil could be in the details here.

Reply:
Another reason is that we have in-memory, ondisc and other cached MSExperiments. How to properly model this multiplicity?

Reply:
maybe not a huge issue but something to think through...

Reply:
Another thing to consider:
our file types support different set of features.
If we make it generic, how to check if the file type behind the input file name is actually providing the data we need?
Before we are sure because we hardcoded filetype. Now we need to check for every replaced position if it is compatible with the file format and otherwise warn/error/document it

Reply:
know those MS Word warnings "you are using a different version ... not all features are supported."
For simpler file types like MGF it is even worse ;)

Reply:
worth to discuss a bit more in detail

Reply:
I don't understand. Can you elaborate?

Reply:
Yes, the options would need to be passed to the function as well. I think it would be too much to store all possible options for all possible file types in the options of the FileHandler class.

Reply:
FileHandler already has `getOptions` and `setOptions` for passing options to the actual type specific `load` call (at least to PeakFiles). I need to think on it a bit, but my first thought is that we simply have the specific file type implementation complain if it gets passed something that doesn't make sense for that type. I don't believe we even do that now with eg. `setForceMQCompatability` and `setForceTPPCompatability`.

Right now we have a bunch of `loadXXX` which correspond to the different objects that we are loading into, I'm playing around with the idea of just having a bunch of overloaded `load` functions in FileHandler that determine based on signature to help with eg. `MapAlignerIdentification` where we use template parameters to specify file type now, but I'm far from sure that it's a good solution.

@timosachsenberg I'm not sure I understand your point about the different types of MSExperiment. Do they have different interfaces?

Reply:
> Isn't what you say implicit through the specific method that is called with the specific target object type? It's not like he is using "load" for everything.

Not sure if I understand. If you use it in a topp tool then you have still the e.g. input restrictions that implicitly tell you with which peak file type it is compatible. 
In the library, in specific algorithms, you might need to introduce restrictions / checks on the file type to make sure that no peak file format is passed that doesn’t make sense. Note that most of our library code assumes the MSExperiment was created from an mzML and might expect certain meta values. This can all be checked and worked around but is not in place, yet.

Reply:
> 

I don't see how that's different from now. MzML files can be perfectly valid and not contain either MS1 or MS2 scans. We handle that at runtime by throwing an error. The only difference with a file-type agnostic load would be that we would need to have the error specify that a specific file type **can not** rather than **does not** have the necessary info.

The other thing to note, is that the `FileHandler.loadXXX(...)` methods allow you to force a file type, so a less aggressive refactoring could still use the FileHandler interface along with forcing the type, and move the filetype includes to the FileHandler class.

Reply:
https://github.com/OpenMS/OpenMS/pull/6780/commits/ca4c0c46487fb3c250f498075521e29874337b5c is a simple example of how that could work,

Reply:
> MzML files can be perfectly valid and not contain either MS1 or MS2 scans.

That is not what I mean. I am talking about formats that can't represent the data  in MSExperiment. These will likely cause trouble.
As I see it: mzML is a serialization format of MSExperiment. other file types kind be loaded into MSExperiment with caveats.

> The only difference with a file-type agnostic load would be that we would need to have the error specify that a specific file type can not rather than does not have the necessary info.

I think forcing the file type could be an option. Otherwise, every place in OpenMS that implicitly assumed that an mzML was used (which is hard to check and confirm) needs to reconsider what types to accept.

Reply:
I spent a while today updating all of our TOPP tests to actually output files with the proper suffix (though with `tmp` as the penumtimate part). I then was able to just get rid of `force`. I think it looks much cleaner now.